pyproject.toml

```diff
@@ -16,6 +16,7 @@ python = "^3.9"
 pydantic = "^2.6.3"
 requests = "^2.31.0"
 python-dateutil = "^2.8.2"
+httpx = "^0.27.2"
```

Comment:
Nice! I haven't used this library before but it looks solid. We're good adding this external dependency to the client. Is seems like `httpx` will replace `requests`, correct? If so, should we consider removing the dependency on `requests` in this PR as well?